presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -1052,7 +1042,7 @@ private boolean canRunMore()
             if (resourceGroupRuntimeInfo.isPresent()) {
                 totalRunningQueries += resourceGroupRuntimeInfo.get().getRunningQueries() + resourceGroupRuntimeInfo.get().getDescendantRunningQueries();
             }
-            int activeWorkerCount = getActiveWorkerCount();
+            int activeWorkerCount = nodeManager.getAllNodes().getActiveWorkerCount();
```

Comment:
How many times per canRunMore invocation was the getActiveWorkerCount called ? Per my understanding, I am not sure if memoizing it is helping since it seems to anyway be called once per invocation. 

Do you also want to fix the duplicate concatenation that happens in InMemoryNodeManager.getallNodes ? 

Maybe we can also fix this issue in the InMemoryNodeManager.getAllNodes by computing the difference there as well ? This whole thing is very confusing due to the use of MultiMap etc and I am wondering if you have considered simplifying it from there.

Reply:
Chatted offline, and resolved.